### PR TITLE
Micrometa Submission

### PR DESCRIPTION
**Pull request by utku-ozturk**
1. Micrometa App's clone and status change fail for non-admin users. (Desired behavior should be like HiGlass.) type/MicroscopyConfiguration's `STATUS_ACL` and `mixins.json/status` schema are updated to solve the issue.
2. If Micrometa item is created via `MicroscopySearchView'`s `Create New` button, then microscope object's ID propert is not set. Backend is updated to set a uuid for ID field.

### CODE REVIEW

File: src/encoded/types/__init__.py [hunk 1]
```diff
@@ -278,6 +281,22 @@ class MicroscopeConfiguration(Item):
...
     schema = load_schema('encoded:schemas/microscope_configuration.json')
+    STATUS_ACL = {
+        'released'              : ALLOW_OWNER_EDIT + ALLOW_CURRENT,
+        'deleted'               : ALLOW_OWNER_EDIT + DELETED,
+        'draft'                 : ALLOW_OWNER_EDIT + ONLY_ADMIN_VIEW,
+        'released to lab'       : ALLOW_OWNER_EDIT + ALLOW_LAB_SUBMITTER_EDIT,
+        'released to project'   : ALLOW_OWNER_EDIT + ALLOW_VIEWING_GROUP_VIEW
+    }
+
+    def _update(self, properties, sheets=None):
```
Comment (aschroed) [inline src/encoded/types/__init__.py:292]: Generally once we add additional functions i.e. _update or calcprops other than a simple display_title we move the type information for an item and collection to it's own types file and write tests.

Comment (utku-ozturk) [reply src/encoded/types/__init__.py:292]: @aschroed Sure, I'll move it to its own types file. (Once integration gets started in November, the pseudo-design was already implemented in init.py and continued as is.)

Comment (burakalver) [reply src/encoded/types/__init__.py:292]: I think deleted remaining editable by owner makes sense. If a user deletes a higlass display or micrometa, they can retrieve it back. If we need it really deleted, we can claim ownership as admin.

File: src/encoded/types/microscopy_configuration.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            microscope = properties.get('microscope')
+            if not microscope.get('ID'):
+                microscope['ID'] = str(uuid4())
+        super(MicroscopeConfiguration, self)._update(properties, sheets)
+
+    @calculated_property(schema={
+        "title": "Display Title",
+        "description": "A calculated title for every object in 4DN",
+        "type": "string"
+    })
```
Comment (j1z0) [inline src/encoded/types/microscopy_configuration.py:53]: Codacy Issue found: [Parameters differ from overridden 'display_title' method](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5525416)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/encoded/types/microscopy_configuration.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5525416&bid=18110672)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/encoded/tests/test_types_microscope_configuration.py  4
- src/encoded/types/microscope_configuration.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5525416&bid=18110672)

File: src/encoded/tests/test_types_microscope_configuration.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import pytest
+from ..schema_formats import is_uuid
```
Comment (j1z0) [inline src/encoded/tests/test_types_microscope_configuration.py:2]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5525416)

File: src/encoded/types/microscope_configuration.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            microscope = properties.get('microscope')
+            if not microscope.get('ID'):
+                microscope['ID'] = str(uuid4())
+        super(MicroscopeConfiguration, self)._update(properties, sheets)
+
+    @calculated_property(schema={
+        "title": "Display Title",
+        "description": "A calculated title for every object in 4DN",
+        "type": "string"
+    })
```
Comment (j1z0) [inline src/encoded/types/microscope_configuration.py:53]: Codacy Issue found: [Parameters differ from overridden 'display_title' method](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5525416)

File: src/encoded/types/__init__.py (hunk 1 above)
Comment (utku-ozturk) [reply src/encoded/types/__init__.py]: Moved MicroscopeConfiguration to its own type file and add new unit tests. ready to be merged.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/encoded/types/microscope_configuration.py  3
- src/encoded/tests/test_types_microscope_configuration.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5525416&bid=18110672)

File: src/encoded/types/microscope_configuration.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
```
Comment (j1z0) [inline src/encoded/types/microscope_configuration.py:53]: Codacy Issue found: [Parameters differ from overridden 'display_title' method](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5525416)

**Pull request merged**